resources/web/wwi/webots.js

```diff
@@ -251,22 +251,86 @@ webots.View = class View {
+                  newSrc = serverUrl.href + 'robot_windows/' + windowName + url.pathname;
+                } catch (_) {
+                  url = null;
+                  newSrc = serverUrl.href + 'robot_windows/' + windowName + '/' + src;
+                }
+                if (url === null || url.origin === serverUrl.origin) {
+                  if (collection[i].tagName === 'SCRIPT') {
+                    collection[i].removeAttribute('src');
+                    collection[i].setAttribute('disabled-src', newSrc);
+                  } else // IMG
+                    collection[i].setAttribute('src', newSrc);
+                } else {
+                  collection[i].setAttribute('disabled-src', src);
+                  collection[i].removeAttribute('src');
+                }
+              }
+            }
+          }
+          let parser = new DOMParser();
+          let doc = parser.parseFromString(data, 'text/html');
+          let serverUrl = new URL(this.server.httpServerUrl);
+          fixSrc(doc.getElementsByTagName('script'), serverUrl);
+          fixSrc(doc.getElementsByTagName('img'), serverUrl);
+          let links = doc.getElementsByTagName('link');
+          for (let i = 0; i < links.length; i++) {
+            if (links[i].rel === 'stylesheet' && links[i].type === 'text/css') {
+              let link = document.createElement('link');
+              for (let j = links[i].attributes.length - 1; j >= 0; j--) {
+                const name = links[i].attributes[j].name;
+                let value = links[i].attributes[j].value;
+                if (name === 'href' && !value.startsWith('https://') && !value.startsWith('http://')) // local css file
+                  value = this.server.httpServerUrl + 'robot_windows/' + windowName + '/' + value;
+                link.setAttribute(name, value);
+              }
+              document.getElementsByTagName('head')[0].appendChild(link);
+            }
+          }
+          const body = doc.getElementsByTagName('body')[0];
+          win.setContent(body.innerHTML);
           MathJax.Hub.Queue(['Typeset', MathJax.Hub, win[0]]);
-          $.get('window/' + windowName + '/' + windowName + '.js', (data) => {
-            eval(data);
-            pendingRequestsCount--;
-            if (pendingRequestsCount === 0)
-              loadFinalize();
-          }).fail(() => {
-            pendingRequestsCount--;
-            if (pendingRequestsCount === 0)
-              loadFinalize();
-          });
+
+          function nodeScriptClone(node) { // this forces the execution of the Javascript code
+            let script = document.createElement('script');
+            script.text = node.innerHTML;
+            for (let i = node.attributes.length - 1; i >= 0; i--) {
+              const name = node.attributes[i].name;
+              const value = node.attributes[i].value;
+              if (name === 'disabled-src') {
+                script.removeAttribute('disabled-src');
```

Comment:
This attribute was set earlier (lines 271 and 276). It is set this way to prevent automatic script execution when creating the DOM objects. As explained in the comment line 304, we call explicitly a function to rename this `src-disabled` into `src` to force the execution of the JavaScript code at this point.

Reply:
Maybe I'm missing something, but the `disabled-src` attribute is set  to the `node` argument but not to the `script` object that has been just created at line 305.
So it seems that rather than just "renaming" `disabled-src` to `src`, in `nodeScriptClone` a new script object is created with the desired `src` value (that was `disabled-src` in the previous object).
It seems that attributes of the `script` object are set at line 318, but this is skipped in case of `disabled-src`.

Reply:
Yes this is on purpose. The idea is that when we receive the script tags and image tags, we need to change the "src" value because the URL is wrong. However, changing the "src" value of a script will cause to execute it which we don't want to do in the temporary doc object because it would be too early and outside the scope of the actual page. We want to do that only when we insert it in the web page. This is why in addition to changing the value of "src", we also rename the field to "disabled-src", so that the script is not executed in the wrong context. However, when adding the script to the web page, it won't be executed, so we need to replace every script element with a new script element which as the correct "src" value (which was stored in the "disabled-src" attribute).

Reply:
But the `script` object doesn't have a `disabled-src` attribute (at least not in any of the robotbenchmark world I tested).
Only the `node` object has the `disabled-src` attribute, is the instruction wrong and should be:
```suggestion
                node.removeAttribute('disabled-src');
```

Reply:
Oh yes... You are right.

Reply:
I believe this line is not even needed at all.